packages/create-redwood-app/template/api/src/lib/db.ts

```diff
@@ -0,0 +1,78 @@
+// See https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/constructor
+// for options.
+
+import { PrismaClient, Prisma } from '@prisma/client'
+import { logger } from './logger'
+
+// To help you identify particularly slow parameter values,
+// For example, Heroku outputs the slowest queries (that take 2 seconds or more)
+const SLOW_QUERY_THRESHOLD = 2 * 1000 // 2 seconds
+
+/*
+ * Determines the type and level of logging.
+ *
+ * See: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#log
+ */
+const logLevels: Prisma.LogLevel[] = ['info', 'warn', 'error'] // add 'query' in dev?
+
+/*
+ * Generates the Prisma Log Definitions from log levels
+ *
+ * @return Prisma.LogDefinition[]
+ */
+const prismaLoggerOptions = (
+  logLevels: Prisma.LogLevel[]
+): Prisma.LogDefinition[] => {
+  return logLevels?.map((level) => {
+    return { emit: 'event', level } as Prisma.LogDefinition
+  })
+}
+
+/*
+ * Generates the Prisma Log Definitions from log levels
+ *
+ * @return PrismaClient
+ */
+const createPrismaClient = (logLevels: Prisma.LogLevel[]) => {
+  const client = new PrismaClient({
+    log: prismaLoggerOptions(logLevels),
+    errorFormat: 'colorless',
+  })
+
+  /*
+   * To use event-based logging:
+   * Set emit to event for a specific log level, such as query
+   * Use the $on() method to subscribe to the event
+   *
+   * https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging
+   */
+
+  logLevels.forEach((level) => {
```

Comment:
Ohh, you need the levels here. Ok. So maybe create an internal const for the levels then :)

Reply:
Yeah it might be nice to abstract this from the user in `api/logger`, unless it's required that the user sees this?

Reply:
> Yeah it might be nice to abstract this from the user in api/logger, unless it's required that the user sees this?

@peterp I thought of that, but if I add something like

`setupPrismaLogevents(client, logLevels)` in `logger` I thought I might have to import Prisma types in `logger` and is seemed odd to for Prisma types into that.

For example: `Prisma.QueryEvent`

Unless I add it to the app's `lib/logger.ts` which should be installed alongside Prisma.

I can try that, though might make better sense and cleaner -- though seeing exactly what Prisma is going to log and how doesn't bother me much.